deployment/common/changeset/link_transfer_timelock.go

```diff
@@ -0,0 +1,144 @@
+package changeset
+
+import (
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/pkg/proposal/timelock"
+
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/common/types"
+)
+
+type LinkTransfer struct {
+	To    common.Address
+	Value *big.Int
+}
+type LinkTransferTimelockRequest struct {
+	Transfers       map[uint64][]LinkTransfer
+	ValidUntil      uint32        // unix time until the proposal will be valid
+	MinDelay        time.Duration // delay for timelock worker to execute the transfers.
+	OverrideRoot    bool
+	StartingOpCount map[uint64]uint64
+	UseMCMS         bool
+}
+
+var _ deployment.ChangeSet[*LinkTransferTimelockRequest] = LinkTransferTimelock
+
+// Validate checks that the LinkTransferTimelockRequest is valid.
+func (cfg LinkTransferTimelockRequest) Validate() error {
```

Comment:
I think the most important thing to check here is the from address has sufficient balance for all the transfers

Reply:
other assertions: 
- To address should be non-empty 
- Chain selectors must be valid 
- Maybe a sanity upper bound on minDelay 
- Maybe reject zero value transfers?

Reply:
Added validations here: https://github.com/smartcontractkit/chainlink/pull/15512/commits/e6ac21a94c96c2457410ef6818ddeaa16e196414